### DISCUSSION TITLE
Can you tell SSURGO vs STATSGO vs RSS from the mukey itself?

### DISCUSSION THREAD

**Discussion by kevinwolz** [ncss-tech/soilDB#336] [Q&A] [answered]
The gNATSGO mukey grid is a mix of SSURGO, STATSGO, and RSS mukeys. Can you tell SSURGO vs STATSGO vs RSS from the mukey itself? For example, by number of digits, value ranges, etc?

**Comment by brownag** [comment, marked as answer]
No, the mukey does not have this information. 

The mukeys map to our internal legend mapunit record IDs, so any changes there are propagated to the final product. When a mapunit is removed from a public legend, that mukey is no longer used. If a new mapunit is added to a legend, it is assigned a new mukey sequentially for the instance of that mapunit in that area. This ultimately means that SSURGO, STATSGO, and RSS mukeys are mixed.
``` r
library(soilDB)

# full range
x <- SDA_query(
  "SELECT MIN(mukey), MAX(mukey) FROM mapunit
   INNER JOIN legend ON legend.lkey = mapunit.lkey"
)
#> single result set, returning a data.frame
x
#>      V1      V2
#> 1 49315 3400375

# STATSGO only
x <- SDA_query(
  "SELECT MIN(mukey), MAX(mukey) FROM mapunit
   INNER JOIN legend ON legend.lkey = mapunit.lkey
   WHERE areasymbol = 'US'"
)
#> single result set, returning a data.frame
x
#>       V1      V2
#> 1 657753 2790295

# STATSGO mukeys are mixed with those from other areas
x <- SDA_query(
  "SELECT COUNT(DISTINCT mukey) FROM mapunit
   INNER JOIN legend ON legend.lkey = mapunit.lkey
   WHERE areasymbol != 'US' AND mukey >= 657753 AND mukey <= 2790295"
)
#> single result set, returning a data.frame
x
#>       V1
#> 1 105658
```

For STATSGO, you can use SDA to look up the mapunit's legend and see what areasymbol it is. All STATSGO mapunits have `legend.areasymbol = 'US'`. You can also use the sastatusmap table to find areas where mapping is not complete. Though, this only has granularity to the soil survey area level, so partially completed SSAs will still need a mapunit-level determination. The only way to know what is RSS is probably to build a lookup table for each state that has one or more raster soil surveys to cross reference against. The spatial and tabular data are available on Box. I have worked on stitching together the RSS grids as a single product, but we don't currently deliver a standalone RSS grid. Assuming you export a gNATSGO grid from the FY24 geodatabase, you know you have all "live" mukeys, so after identifying STATSGO mapunits you could assume remaining unknown mukeys were RSS.

**Reply by brownag** [reply]
Note also that if you use the gNATSGO File Geodatabase, you can use the `legend.legenddesc` column to identify the `"Raster Soil Map Legend"` legends

**Reply by kevinwolz** [reply]
Thanks, this is very helpful! I am trying to do this locally with the gNATSGO mukey raster and the CSV tabular data. However, it seems like the gNATSGO CSV tabular data does not contain the mapunit and legend tables. Is this expected? Those tables _are_ in the gSSURGO CSV tabular data.

**Reply by brownag** [reply]
> However, it seems like the gNATSGO CSV tabular data does not contain the mapunit and legend tables. Is this expected? 

That is not expected. I just downloaded a fresh FY23 copy and both mapunit.csv and legend.csv are present for gNATSGO.
![image](https://github.com/ncss-tech/soilDB/assets/20842828/d829c648-d777-447b-9f29-6c456267ceed)

**Reply by kevinwolz** [reply]
Okay I re-downloaded it and they are there now! I think something must have timed out while downloading and it didn't get to those.
